### DISCUSSION TITLE
Quick search totally broken after upgrade

### DISCUSSION THREAD

**Discussion by kotso** [netbox-community/netbox#11809] [General]
Hello, 

Upgraded yesterday to latest version of NetBox v3.4.5, was using 3.3.x before for long time and noticed that search is totally broken or Im doing something wrong.

<img width="1059" alt="Screenshot 2023-02-23 at 10 57 38" src="https://user-images.githubusercontent.com/1928528/220839252-bfce96c7-d8ed-4cb2-9f78-46a652f3dca5.png">
<img width="1059" alt="Screenshot 2023-02-23 at 10 57 18" src="https://user-images.githubusercontent.com/1928528/220839261-d6a12b22-c9e7-48ff-b35d-0c23282bf598.png">

tried exact match, tried start with, regex, nothing found.

Am I doing something wrong here?

**Comment by kotso** [comment]
Checked same on demo, it's working there. 

Check docs, did not find any special upgrade instructions or anything. Using docker version of NetBox. https://github.com/netbox-community/netbox-docker

**Comment by kotso** [comment]
did some more digging, only thing I was able to find is journal entry by partial match.

**Comment by kkthxbye-code** [comment]
Please see https://github.com/netbox-community/netbox-docker/issues/947

You can manually run the `reindex --lazy` command to fix it.

Edit:

For others with the same problem, if you haven't touched (created or updated) any ipaddress, prefixes or aggregates, you can use the `reindex --lazy` command. If you did touch any of them, you will need to run the `reindex` command without the `--lazy` option which will regenerate the entire search index and might take up to 10 minutes depending on instance size.

**Reply by Simperdy** [reply]
Thanks man this helped me a lot.
We also had a Problem with the global search in Docker Netbox. (in Kubernetes)
After a manage.py reindex without the --lazy the search finaly works again and we can find our prefixes.

**Comment by kotso** [comment]
@kkthxbye-code thank you very much, it did the trick.

If I get it correctly next update of docker-netbox will fix it and no need to manually run that.

**Reply by kkthxbye-code** [reply]
You would still have to run the command manually from within the container. There's instructions in the wiki in the netbox-docker repo.

**Comment by jeremystretch** [comment]
Please note that the two screenshots above are of two entirely unrelated search functions. The first is the object-specific quick search (e.g. `/ipam/ip-addresses/?q=foo`); the second is the global search feature.

**Comment by kotso** [comment]
@jeremystretch sure, I just posted it to show that object really exists but I could not find it from global search.

@kkthxbye-code most of issues has been fixed since running index update but still somethings can't be found.

I run manually reindex again but no luck, I tried to dig further and found strange thing.

```
  dcim.device... Skipping (found 10 existing).
  dcim.devicebay... Skipping (found 100 existing).
  dcim.devicerole... Skipping (found 32 existing).
  dcim.devicetype... Skipping (found 101 existing).
  dcim.frontport... Skipping (found 196 existing).
  dcim.interface... Skipping (found 34294 existing).
```

that's part of index update script output.

What I find strange there is numbers. I have 906 devices in Netbox currently, but somehow it says only 10 (if I get it correctly that found XXX means number of items script things should be in cache) but number of interfaces looks correct. (one that's 34K.)

First thought was that it's getting some single status only but I was unable to find any status where I would have exactly 10 devices. 

As an clue, I'm using custom statuses also on dcim.device model that was introduced couple releases ago, if it may be somehow related.

**Reply by kkthxbye-code** [reply]
We didn't touch the device index in the update. But you can run reindex without the lazy option to recreate all indexes.

**Comment by kotso** [comment]
@kkthxbye-code running index without lazy did the trick, thanks a lot. 🙏

**Comment by greinick** [comment]
Hi, I'v also a question regarding the quick search e.g. in the Devices submenu:

I'm searching for e.g. "Nexus" and I know that I do have some in Netbox with the same syntax and labels etc. Only one result is returned. All other devices are not.

I did the reindexing, but no change. What might be wrong? I'm using NB 3.4.10

Regards & thanks.

**Reply by kkthxbye-code** [reply]
The quick search for devices uses the following filter:

```
Q(name__icontains=value) |
Q(serial__icontains=value.strip()) |
Q(inventoryitems__serial__icontains=value.strip()) |
Q(asset_tag__icontains=value.strip()) |
Q(comments__icontains=value) |
Q(primary_ip4__address__startswith=value) |
Q(primary_ip6__address__startswith=value)
```

It's likely that you are expecting it to search a field it is not actually searching, like the device type name.

**Reply by greinick** [reply]
Cool thanks, can I change/edit the filter? Or should I look for an other way to search. (Would be ok for me)

**Reply by kkthxbye-code** [reply]
No, you cannot change the filter. If you are looking for something specific, like all the devices of a specific devicetype, you should use the field filters instead of the generic quick search.

**Comment by hoangpbc** [comment]
./manage reindex --lazy work like a charm, thank alot!
i'm now on version 4.7.
